dev/deps/spark-deps-hadoop-2.7

```diff
@@ -149,10 +149,11 @@ lz4-java-1.6.0.jar
 machinist_2.12-0.6.8.jar
 macro-compat_2.12-1.1.1.jar
 mesos-1.4.0-shaded-protobuf.jar
-metrics-core-3.1.5.jar
-metrics-graphite-3.1.5.jar
-metrics-json-3.1.5.jar
-metrics-jvm-3.1.5.jar
+metrics-core-4.1.1.jar
+metrics-graphite-4.1.1.jar
+metrics-jmx-4.1.1.jar
```

Comment:
Tiny nit, could you add an entry to `LICENSE-binary` with this new dependency? you'll see where it goes